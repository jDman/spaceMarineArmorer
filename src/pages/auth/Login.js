import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import Auth from './Auth';
import Input from '../../components/form/input/Input';

class Login extends Component {
  state = {
    loading: false,
    loginFormIsValid: true,
    loginForm: {
      email: {
        elementType: 'input',
        elementConfig: {
          type: 'email',
          name: 'email',
          placeholder: 'you@email.com',
          required: true
        },
        value: '',
        valid: false,
        validation: { required: true },
        touched: false
      },
      password: {
        elementType: 'input',
        elementConfig: {
          type: 'password',
          name: 'password',
          placeholder: 'Enter you password...',
          required: true
        },
        value: '',
        valid: false,
        validation: { required: true },
        touched: false
      }
    }
  };

  inputChangedHandler(event, identifier) {
    const updatedLoginForm = {
      ...this.state.loginForm
    };

    const updatedLoginDetail = { ...updatedLoginForm[identifier] };

    updatedLoginDetail.value = event.target.value;
    updatedLoginDetail.touched = true;

    if (updatedLoginDetail.validation) {
      updatedLoginDetail.valid = this.checkValidity(
        updatedLoginDetail.value,
        updatedLoginDetail.validation
      );
    }

    updatedLoginForm[identifier] = updatedLoginDetail;

    let formIsValid = true;

    for (let key in updatedLoginForm) {
      const isDetailValid = updatedLoginForm[key].valid;

      if (isDetailValid !== null && isDetailValid !== undefined) {
        formIsValid = isDetailValid && formIsValid;
      } else {
        formIsValid = true && formIsValid;
      }
    }

    this.setState({
      loginForm: updatedLoginForm,
      loginFormIsValid: formIsValid
    });
  }

  login(event) {
    event.preventDefault();
  }

  render() {
    const loginFormElements = [];
    const loginForm = this.state.loginForm;

    for (let key in loginForm) {
      loginFormElements.push({
        id: key,
        config: loginForm[key]
      });
    }

    return (
      <Auth>
        <form onSubmit={event => this.login(event)}>
          {loginFormElements.map(formElement => {
            return (
              <Input
                key={formElement.id}
                elementType={formElement.config.elementType}
                elementConfig={formElement.config.elementConfig}
                value={formElement.config.value}
                invalid={!formElement.config.valid}
                touched={formElement.config.touched}
                hasValidation={formElement.config.validation}
                changed={event =>
                  this.inputChangedHandler(event, formElement.id)
                }
              />
            );
          })}
        </form>
      </Auth>
    );
  }
}

export default withRouter(Login);
