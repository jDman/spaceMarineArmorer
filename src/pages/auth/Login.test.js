import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Login from './Login';

test('renders Login page', () => {
  const component = renderer.create(
    <Router>
      <Login />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
