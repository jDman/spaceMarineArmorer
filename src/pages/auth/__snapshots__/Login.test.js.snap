// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page 1`] = `
<section
  className="auth-form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="Input"
    >
      <label
        className="Label"
      />
      <input
        className="InputElement"
        name="email"
        onChange={[Function]}
        placeholder="you@email.com"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="Input"
    >
      <label
        className="Label"
      />
      <input
        className="InputElement"
        name="password"
        onChange={[Function]}
        placeholder="Enter you password..."
        required={true}
        type="password"
        value=""
      />
    </div>
  </form>
</section>
`;
